### PR TITLE
Extension Level

### PR DESCRIPTION
**Pull request by duane701**
Adding a Pull Request in order to comment on the code.

### CODE REVIEW

File: Brokkr/src/main/cpp/RobotContainer.cpp [hunk 1]
```diff
@@ -21,6 +22,8 @@ RobotContainer::RobotContainer() {
  frc::SmartDashboard::PutData("ArmPosition Mid", new ArmPosition(mArm, 0));
  frc::SmartDashboard::PutData("ArmPosition High", new ArmPosition(mArm, 0));
  frc::SmartDashboard::PutData("ArmPosition Shelf", new ArmPosition(mArm, 0));
+ frc::SmartDashboard::PutData("Wow", new SetArmPostitionForDistance(mArm, NodeLevel::HybridLevel));
```
Comment (duane701) [inline Brokkr/src/main/cpp/RobotContainer.cpp]: Really? "Wow"? 😉 

Probably should have a more descriptive label.

File: Brokkr/src/main/include/commands/SetArmPostitionForDistance.h [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+#include "subsystems/Arm.h"
+
+/**
+ * An example command.
+ *
+ * <p>Note that this extends CommandHelper, rather extending CommandBase
+ * directly; this is crucially important, or else the decorator functions in
+ * Command will *not* work!
+ */
+  enum class NodeLevel {HybridLevel, MiddleNodeLevel, UpperNodeLevel};
```
Comment (duane701) [inline Brokkr/src/main/include/commands/SetArmPostitionForDistance.h:19]: Naming consistency. 

```
  enum class NodeLevel {HybridNodeLevel, MiddleNodeLevel, UpperNodeLevel};
```

**Review by duane701 — COMMENTED** [review]
More comments as I read more.

File: Brokkr/src/main/cpp/commands/SetArmPostitionForDistance.cpp [hunk 3]
```diff
@@ -18,57 +27,74 @@ namespace{
     return angle = atan(height/distance);
   }
 
-  double GetNodeDistance(const NodeLevel& NodeLevel, double distance)
+    const double kHybridHeight = 0;
```
Comment (duane701) [inline Brokkr/src/main/cpp/commands/SetArmPostitionForDistance.cpp:30]: Naming consistency.

```
    const double kHybridNodeHeight = 0;
```

File: Brokkr/src/main/cpp/RobotContainer.cpp (hunk 1 above)
Comment (OneGneissGuy) [reply Brokkr/src/main/cpp/RobotContainer.cpp]: Is this referring to the "Hybrid" position?

File: Brokkr/src/main/cpp/commands/SetArmPostitionForDistance.cpp [hunk 4]
```diff
@@ -18,57 +27,74 @@ namespace{
...
+    const double kUpperNodeHeight = 46;
+
+  double GetNodeDistance(const NodeLevel& NodeLevel, double AngleTargetOffset)
   {
+    double distance = 0;
     switch (NodeLevel)
     {
     case NodeLevel::HybridLevel:
-      return GetDistance(0, distance);
+      distance = GetCameraDistance(kHybridHeight, AngleTargetOffset);
```
Comment (duane701) [inline Brokkr/src/main/cpp/commands/SetArmPostitionForDistance.cpp:40]: `GetCameraDistance` is called for both GetNodeDistance and GetNodeAngle. 

Consider optimizing to call `GetCameraDistance` only once.

We should make sure that the math is correct _BEFORE_ optimizing.

File: Brokkr/src/main/cpp/commands/SetArmPostitionForDistance.cpp [hunk 5]
```diff
@@ -6,6 +6,15 @@
 #include <cmath>
 
 namespace{
+
+  double GetCameraDistance(double NodeHeight, double AngleTargetOffset)
+  {   
+    const double kCameraAngle = 45*(3.14152653589/180);
```
Comment (duane701) [inline Brokkr/src/main/cpp/commands/SetArmPostitionForDistance.cpp]: Can we work in radians?

Comment (duane701) [reply Brokkr/src/main/cpp/commands/SetArmPostitionForDistance.cpp]: Answering my own questions. The camera returns yaw, pitch, and roll in degrees.

We can choose to convert those measurements to radians as they are read or convert from degrees to radians whenever we want to make calculations.

My preference would be to do all of the math in radians and convert inputs and outputs as needed.

File: Brokkr/src/main/cpp/commands/SetArmPostitionForDistance.cpp [hunk 6]
```diff
@@ -18,57 +27,74 @@ namespace{
     return angle = atan(height/distance);
   }
 
-  double GetNodeDistance(const NodeLevel& NodeLevel, double distance)
+    const double kHybridHeight = 0;
+    const double kMiddleNodeHeight = 34;
```
Comment (duane701) [inline Brokkr/src/main/cpp/commands/SetArmPostitionForDistance.cpp:31]: Seems like there are 2 middle heights and 2 upper heights. 

```
    const double kMiddleConeNodeHeight = 34;
    const double kMiddleCubeNodeHeight = 34;
```

File: Brokkr/src/main/cpp/commands/SetArmPostitionForDistance.cpp [hunk 7]
```diff
@@ -18,57 +27,74 @@ namespace{
...
+    const double kUpperNodeHeight = 46;
```
Comment (duane701) [inline Brokkr/src/main/cpp/commands/SetArmPostitionForDistance.cpp:32]:
```
    const double kUpperConeNodeHeight = 46;
    const double kUpperCubeNodeHeight = 46;
```

File: Brokkr/src/main/cpp/commands/SetArmPostitionForDistance.cpp (hunk 6 above)
Comment (duane701) [reply Brokkr/src/main/cpp/commands/SetArmPostitionForDistance.cpp:31]: Is there an assumption that the difference between these heights doesn't matter?

**Pull request merged**